core/data/types.rng

```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+
+<grammar xmlns="http://relaxng.org/ns/structure/1.0"
+         datatypeLibrary="http://www.w3.org/2001/XMLSchema-datatypes">
+
+    <define name="libero.types.id">
+
+        <data type="string">
+            <param name="pattern">\s*[A-Za-z0-9\-._]+\s*</param>
```

Comment:
Good practice to allow whitespace either side?

Reply:
For implementations, I take it this white space will be trimmed? If so, will it be the case that all text nodes are trimmed, unless stated?

Reply:
I think we'll need to specify how to handle whitespace in the data model spec.

Reply:
I guess the use-case for white spaces would be to have formatting like this:
```xml
<id>
  my-id
</id>
```
Automatic XML formatting might add extra whitespaces where there are child elements. e.g.
```xml
<title>
  My
  <i>italic</i>
  title
</title>
```
In general I would opt for not trimming whitespaces to avoid special cases where we do want to have whitespaces. Although the example with the _title_ would suggest trimming whitespaces might be a good option.

(I understand this is related to the _id_ but I guess we don't want to have different whitespace handling per field)